### DISCUSSION TITLE
Transistors and transmission gates: should we remove them?

### DISCUSSION THREAD

**Discussion by BFH-ktt1** [logisim-evolution/logisim-evolution#841] [General]
From the original logisim version there are still transistors and transmission gates in the wiring library. These are analog components that are not completely simulated correctly in logisim (due to it's digital simulation engine). 
As these two components only simulate "correctly" in a certain way of connection, the usage is quite limited, and if not connected correctly, even wrong.
I am quite in favor of suppressing these these two components and would like to have your feed-back.

**Comment by MarcinOrlowski** [comment]
I think that this question would be easier answered once we got a roadmap. The trick is this. If we know where we are heading, it is easier to say given feature fits or can be removed. Also, we shall somehow figure out if backward incompatible changes are really a big deal or not much of a problem. If we nuke tons of libraries, well, then this is problematic. But if there's one lib AND it is actively maintainer AND we can coordinate the change, then if the change is beneficial, we should go that way. Anyway, this all leads to ROADMAP (#695). I think it should really be somehow discussed first.

**Comment by mbaillif** [comment]
I still find these transistor symbols interesting. 
To present the construction of MOS logic circuits and the serial circuit aspect of the "AND" function and the parallel circuit of the "OR" function.
We can see that in automation.

We can't realize all what is in this article, but the logic functions work. 
https://wiki.analog.com/university/courses/electronics/electronics-lab-28.

We can also observe that the limitations of simulating logisim in this context are almost the same as for real logic gates.
Experience has shown that what they simulate best is that they have to be connected correctly :-))

In the same theme we are able to present with the right buffers and parameters diode-resistor gates but it is only the diode symbol that is missing.

In my mind and I think also for most of the logisim users it is well understood that we simulate only the 1/0 logic and not something else so there is no confusion.

**Reply by maehne** [reply]
I agree with @mbaillif. The transistor components are useful to explain the working of NMOS/PMOS/CMOS gates in an easy visual way. Of course, they have their limitations and require attention when wiring them together. I highlight that when I discuss it with my students. One advantage I see over the demonstration of these circuits in a SPICE simulator, such as LTspice, is that you directly can see the signal values in the schematic. So, I am in favour of keeping these components and checking in the documentation whether their limitations need to be more strongly highlighted.
